src/middlewared/middlewared/plugins/chart_releases_linux/attachments.py

```diff
@@ -0,0 +1,20 @@
+from middlewared.common.ports import PortDelegate
+
+
+class ChartReleasePortDelegate(PortDelegate):
+
+    name = 'applications'
+    namespace = 'chart.release'
+    title = 'Applications'
+
+    async def get_ports(self):
+        ports = []
+        for chart_release in await self.middleware.call('chart.release.query'):
+            for port in chart_release['used_ports']:
+                ports.append(port['port'])
```

Comment:
What if we change return schema to `{8080: "Application 1", 8081: "Application 2"}`? That way we'll be able to report more specific error message (`This port is used by Application 1` instead of `This port is used by Applications`).